github/repos_prereceive_hooks.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2013 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// PreReceiveHook represents a GitHub pre-receive hook for a repository.
+type PreReceiveHook struct {
+	ID          *int64  `json:"id,omitempty"`
+	Name        *string `json:"name,omitempty"`
+	Enforcement *string `json:"enforcement,omitempty"`
+	ConfigURL   *string `json:"configuration_url,omitempty"`
+}
+
+func (h PreReceiveHook) String() string {
+	return Stringify(h)
+}
+
+// ListPreReceiveHooks lists all pre-receive hooks for the specified repository.
+//
+// GitHub API docs: https://developer.github.com/enterprise/2.13/v3/repos/pre_receive_hooks/#list-pre-receive-hooks
+func (s *RepositoriesService) ListPreReceiveHooks(ctx context.Context, owner, repo string, opt *ListOptions) ([]*PreReceiveHook, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/pre-receive-hooks", owner, repo)
+	u, err := addOptions(u, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypePreReceiveHooksPreview)
+
+	var hooks []*PreReceiveHook
+	resp, err := s.client.Do(ctx, req, &hooks)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return hooks, resp, nil
+}
+
+// GetPreReceiveHook returns a single specified pre-receive hook.
+//
+// GitHub API docs: https://developer.github.com/enterprise/2.13/v3/repos/pre_receive_hooks/#get-a-single-pre-receive-hook
+func (s *RepositoriesService) GetPreReceiveHook(ctx context.Context, owner, repo string, id int64) (*PreReceiveHook, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/pre-receive-hooks/%d", owner, repo, id)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypePreReceiveHooksPreview)
+
+	hook := new(PreReceiveHook)
+	resp, err := s.client.Do(ctx, req, hook)
+	return hook, resp, err
```

Comment:
I think we would like the error checking to remain consistent, like your function above:

```go
if err != nil {
	return nil, resp, err
}

return hook, resp, nil
```

We think this is easier to read and reason about.

If you copied this pattern from somewhere else in this repo, we should probably change that one too.

Reply:
Yes, this file was a copy-modify of `repos_hooks.go`, so these issues are also copied from there. Will make the changes in this PR and open a separate PR to fix the occurrences in `repos_hooks.go`.